Lib/logging/handlers.py

```diff
@@ -1532,6 +1532,19 @@ def __init__(self, queue, *handlers, respect_handler_level=False):
         self._thread = None
         self.respect_handler_level = respect_handler_level
 
+    def __enter__(self):
+        """
+        For use as a context manager. Starts the listener.
+        """
+        self.start()
```

Comment:
Should we prevent multiple `__enter__`? (start() does not prevent it but should we do it in a follow-up PR?)

Reply:
Do you mean prevent nested starts? As in start called while it's running? 

.. if so: I think so.

If not: it should be safe to call start then stop then start again later.

Reply:
Added.